components/merkledb/tests/proof_map_index.rs

```diff
@@ -68,15 +71,16 @@ where
     Ok(())
 }
 
-fn check_map_multiproof<T, K, V>(
-    proof: &MapProof<K, V>,
+fn check_map_multiproof<T, K, V, S>(
+    proof: &MapProof<K, V, S>,
     keys: BTreeSet<&K>,
-    table: &ProofMapIndex<T, K, V>,
+    table: &ProofMapIndexBase<T, K, V, S>,
```

Comment:
This test sort of relies on the assumption that keys are not transformed. Hence, there are special test cases for keys with limited key space (`TestParams.key_bytes` field), which (if the keys are not transformed) leads to a particular structure of the Merkle Patricia tree. I'd suggest to use untransformed keys (e.g., `Hash`) in this test.